redis/src/cluster_async/mod.rs

```diff
@@ -1165,18 +1174,16 @@ where
             }
             InternalSingleNodeRouting::SpecificNode(route) => read_guard
                 .connection_for_route(&route)
-                .map_or(ConnectionCheck::Nothing, ConnectionCheck::Found),
-            InternalSingleNodeRouting::Random => ConnectionCheck::Nothing,
+                .map_or(ConnectionCheck::RandomConnection, ConnectionCheck::Found),
```

Comment:
We should note that if the user do some request with a specific node route, and this node isn't found, we will route it to a random node. If the command is key-based, there's no harm. however, if the command is for example flushall, config set, or other management command - it will be transparent to the user.
We might want to add a warning here

Reply:
done